samples/README.md

```diff
@@ -18,3 +22,135 @@ for that code, and they always have the option not to.
+    module-level defaulting.
+
+## Restrictions
+
+Files must be UTF-8 encoded.
+
+Files must contain only printable ASCII characters and `\n`.
+
+Files must be compatible with Java 8.
+
+Files must compile without error using stock javac.
+
+Files must not depend on any classes other than the JSpecify annotations. This
+includes the Java platform APIs. Exception: Files may use `java.lang.Object`,
+but they still must not use its methods.
+
+> For example, files may use `Object` as a bound, parameter, or return type.
+
+Files should avoid depending on the presence of absence of "smart" checker
+features, such as:
+
+-   looking inside the body of a method to determine what parameters it
+    dereferences or what it returns
+
+    -   To that end, prefer abstract methods when practical.
+
+-   flow-sensitive typing
+
+We also encourage writing files that demonstrate individual behaviors in
+isolation. For example, we encourage writing files to minimize how much they
+rely on type inference -- except, of course, for any files explicitly intended
+to demonstrate type inference.
+
+## What sample inputs demonstrate
+
+Sample inputs demonstrate 2 cases:
+
+1.  JSpecify annotations are applied in a way that is
+    [structurally illegal](https://docs.google.com/document/d/15NND5nBxMkZ-Us6wz3Pfbt4ODIaWaJ6JDs4w6h9kUaY/edit#heading=h.ib00ltjpj1xa).
+
+    <!-- TODO: Are we happy with the term "illegal?" If so, use it in the comment text instead of "paradox?" -->
+
+2.  The second case is more nuanced: Based on JSpecify annotations and rules,
+    the code's types can all be augmented with nullness information. We could
+    then apply normal JLS rules to those types. (For example, the value of a
+    `return` statement must be convertible to the method's return type.) We
+    could adapt those rules to use JSpecify rules for subtyping. Based on that,
+    we could identify type checks that would fail. (For example, `return null`
+    in a method that must return a non-nullable type.)
+
+    <!-- TODO: Update links to point to the markup-format spec and glossary. -->
+
+## Syntax
+
+We define a format for Java comments to identify code that demonstrates the
+cases above.
+
+A comment on a given line provides information about the following line.
+
+Such a comment contains one of 3 special sequences:
+
+-   `jspecify_nullness_paradox`: for case 1 above
+
+-   `jspecify_nullness_mismatch`: for certain instances of case 2
+
+-   `jspecify_nullness_not_enough_information`: for certain instances of case 2
+
+The difference between the "mismatch" and "not enough information" cases arises
+from
+[unspecified nullness](https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=id.xb9w6p3ilsq3).
+The rough idea is that a type with unspecified nullness means: "When this code
+is annotated for nullness, either the type should have `@Nullable` or it, or it
+shouldn't."
```

Comment:
Looks like a typo around "or it, or it shouldn't", but can't decipher what was the intended meaning

Reply:
The first "or" should be an "on."

I'm still not sure if I put this clearly, so let me know.